core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -1935,22 +1929,19 @@ private long loadDirectoryMetadataAndJournal(LockedInodePath inodePath,
   }
 
   /**
-   * Loads metadata for the path if it is (non-existing || (persisted && load direct children is
-   * set).
+   * Loads metadata for the path if it is (non-existing || load direct children is set).
    *
    * @param inodePath the {@link LockedInodePath} to load the metadata for
    * @param options the load metadata options
    */
-  // TODO(peis): Add a unit test for this function.
   private long loadMetadataIfNotExistAndJournal(LockedInodePath inodePath,
       LoadMetadataOptions options) {
     boolean inodeExists = inodePath.fullPathExists();
     boolean loadDirectChildren = false;
     if (inodeExists) {
       try {
         Inode<?> inode = inodePath.getInode();
-        loadDirectChildren =
-            inode.isDirectory() && inode.isPersisted() && options.isLoadDirectChildren();
+        loadDirectChildren = inode.isDirectory() && options.isLoadDirectChildren();
```

Comment:
I have modified the unittest i added to check this persisted bit also.

That the current loadMetadata implementation is not trying to synchronize with ufs.  

This is the current behavior:

1. getStatus does nothing if the directory/file exists.
2. listStatus on an existing file (non-persisted) does nothing (even if it exists on ufs). It won't reset the persisted bit.
3. listStatus on an existing directory. It will try to load its children if it has never done this before or the user forces loading metadata. It will also reset the directory's persistence stats and  permission bits.  But it won't reset the file's permission bit and persistence state because we return early. The inconsistency between file and directory is for efficiency.  In order to address this (if we want), we can add another bit in file Inode to indicate that we have loaded metadata for this file or not.